File: static/js/health.js
// health.js - extracted from Health.html
            // Health data storage (in memory - persists during session)
            let healthRecords = [
                { date: '2025-11-01', weight: 75.2, bodyFat: 14.2, performance: 'Good energy, completed full workout' },
                { date: '2025-10-15', weight: 75.8, bodyFat: 14.5, performance: 'Slightly tired, reduced intensity' },
                { date: '2025-10-01', weight: 76.1, bodyFat: 14.8, performance: 'Excellent performance, PR on pull-ups' },
                { date: '2025-09-15', weight: 76.5, bodyFat: 15.1, performance: 'Recovery day, light mobility work' },
                { date: '2025-09-01', weight: 77.0, bodyFat: 15.5, performance: 'Started new program, feeling good' }
            ];

            // Health functionality
            document.addEventListener('DOMContentLoaded', function() {
                const addHealthRecordBtn = document.getElementById('addHealthRecordBtn');
                const addRecordModal = document.getElementById('addRecordModal');
                const closeRecordModal = document.getElementById('closeRecordModal');
                const cancelRecordBtn = document.getElementById('cancelRecordBtn');
                const saveRecordBtn = document.getElementById('saveRecordBtn');
                const exportDataBtn = document.getElementById('exportDataBtn');
                
                // Set current date in record form
                const today = new Date().toISOString().split('T')[0];
                document.getElementById('recordDate').value = today;
                
                // Open modal
                addHealthRecordBtn.addEventListener('click', function() {
                    addRecordModal.style.display = 'flex';
                });
                
                // Close modal functions
                function closeRecordModalFunc() {
                    addRecordModal.style.display = 'none';
                    // Reset form
                    document.getElementById('recordWeight').value = '';
                    document.getElementById('recordBodyFat').value = '';
                    document.getElementById('recordPerformance').value = '';
                }
                
                closeRecordModal.addEventListener('click', closeRecordModalFunc);
                cancelRecordBtn.addEventListener('click', closeRecordModalFunc);
                
                // Close modal when clicking outside
                addRecordModal.addEventListener('click', function(e) {
                    if (e.target === addRecordModal) {
                        closeRecordModalFunc();
                    }
                });
                
                // Save health record
                saveRecordBtn.addEventListener('click', function() {
                    const weight = document.getElementById('recordWeight').value;
                    const bodyFat = document.getElementById('recordBodyFat').value;
                    const performance = document.getElementById('recordPerformance').value;
                    const date = document.getElementById('recordDate').value;
                    
                    if (!weight || !date) {
                        alert('Please fill in required fields!');
                        return;
                    }
                    
                    // Add to health records array
                    healthRecords.push({
                        date: date,
                        weight: parseFloat(weight),
                        bodyFat: bodyFat ? parseFloat(bodyFat) : null,
                        performance: performance || 'No notes'
                    });
                    
                    alert(`Health record saved!\nWeight: ${weight}kg\nBody Fat: ${bodyFat || 'N/A'}%\nDate: ${date}`);
                    closeRecordModalFunc();
                    
                    // Update calendar to show measured days
                    updateCalendarMeasuredDays();
                });
                
                // Export data to text file
                exportDataBtn.addEventListener('click', function() {
                    if (healthRecords.length === 0) {
                        alert('No health records to export!');
                        return;
                    }
                    
                    // Create text content
                    let textContent = 'CALISTHENICS HEALTH DATA\n';
                    textContent += '========================\n\n';
                    
                    // Add basic info
                    const weight = document.getElementById('weight').value;
                    const height = document.getElementById('height').value;
                    const age = document.getElementById('age').value;
                    const gender = document.getElementById('gender').value;
                    
                    textContent += 'BASIC INFORMATION:\n';
                    textContent += `Weight: ${weight} kg\n`;
                    textContent += `Height: ${height} cm\n`;
                    textContent += `Age: ${age} years\n`;
                    textContent += `Gender: ${gender}\n`;
                    textContent += `BMI: ${calculateBMI(weight, height)}\n\n`;
                    
                    // Add health records
                    textContent += 'HEALTH RECORDS:\n';
                    textContent += 'Date,Weight(kg),BodyFat(%),Performance\n';
                    
                    // Sort records by date (newest first)
                    const sortedRecords = [...healthRecords].sort((a, b) => new Date(b.date) - new Date(a.date));
                    
                    sortedRecords.forEach(record => {
                        textContent += `${record.date},${record.weight},${record.bodyFat || 'N/A'},"${record.performance}"\n`;
                    });
                    
                    // Create and download file
                    const blob = new Blob([textContent], { type: 'text/plain' });
                    const url = URL.createObjectURL(blob);
                    const a = document.createElement('a');
                    a.href = url;
                    a.download = `calisthenics_health_data_${new Date().toISOString().split('T')[0]}.txt`;
                    document.body.appendChild(a);
                    a.click();
                    document.body.removeChild(a);
                    URL.revokeObjectURL(url);
                    
                    alert('Health data exported successfully! Check your downloads folder.');
                });
                
                // Plan item functionality
                document.querySelectorAll('.plan-action-btn.complete').forEach(btn => {
                    btn.addEventListener('click', function() {
                        const planItem = this.closest('.plan-item');
                        planItem.classList.toggle('completed');
                        const icon = this.querySelector('i');
                        if (planItem.classList.contains('completed')) {
                            icon.className = 'fas fa-undo';
                            this.title = 'Mark as incomplete';
                        } else {
                            icon.className = 'fas fa-check';
                            this.title = 'Mark as complete';
                        }
                    });
                });
                
                document.querySelectorAll('.plan-action-btn.delete').forEach(btn => {
                    btn.addEventListener('click', function() {
                        const planItem = this.closest('.plan-item');
                        if (confirm('Are you sure you want to delete this health plan item?')) {
                            planItem.style.opacity = '0';
                            planItem.style.transform = 'translateX(-20px)';
                            setTimeout(() => {
                                planItem.remove();
                            }, 300);
                        }
                    });
                });
                
                // Calendar day click functionality
                document.querySelectorAll('.calendar-day').forEach(day => {
                    day.addEventListener('click', function() {
                        if (this.classList.contains('measured') || this.classList.contains('today')) {
                            const dayNumber = this.textContent;
                            alert(`Health measurements recorded for November ${dayNumber}, 2025`);
                        } else {
                            alert(`No health measurements recorded for November ${this.textContent}, 2025`);
                        }
                    });
                });
                
                // Initialize calendar
                updateCalendarMeasuredDays();
            });
            
            function calculateBMI(weight, height) {
                // weight in kg, height in cm
                const heightM = parseFloat(height) / 100;
                const bmi = parseFloat(weight) / (heightM * heightM);
                return bmi.toFixed(1);
            }
            
            function updateCalendarMeasuredDays() {
                // Get all dates from health records
                const recordedDates = healthRecords.map(record => {
                    const date = new Date(record.date);
                    return date.getDate().toString();
                });
                
                // Add 'measured' class to calendar days that have records
                document.querySelectorAll('.calendar-day').forEach(day => {
                    const dayNumber = day.textContent;
                    if (recordedDates.includes(dayNumber)) {
                        day.classList.add('measured');
                    }
                });
            }

File: app/routes/health.py
import io
from flask import Blueprint, render_template, request, redirect, url_for, flash, Response
from datetime import date
from app import db
from app.models.health import HealthRecord, HealthPlanItem

health_bp = Blueprint('health', __name__)


@health_bp.route('/')
def view():
    latest = HealthRecord.query.order_by(HealthRecord.date.desc()).first()
    records = HealthRecord.query.order_by(HealthRecord.date.desc()).all()
    plan_items = HealthPlanItem.query.order_by(HealthPlanItem.created_at.desc()).all()

    chart_data = []
    for r in reversed(records):
        if r.weight:
            chart_data.append({'date': r.date.strftime('%b %d') if r.date else '', 'weight': r.weight})

    return render_template('health.html',
        active_page='health',
        latest=latest,
        records=records,
        plan_items=plan_items,
        chart_data=chart_data,
    )


@health_bp.route('/add', methods=['POST'])
def add_record():
    weight = request.form.get('weight', type=float)
    height = request.form.get('height', type=float)
    age = request.form.get('age', type=int)
    gender = request.form.get('gender', '')
    body_fat = request.form.get('body_fat', type=float)
    record_date_str = request.form.get('date', '')

    record_date = date.fromisoformat(record_date_str) if record_date_str else date.today()
    record = HealthRecord(date=record_date, weight=weight, height=height, age=age, gender=gender, body_fat=body_fat)
    record.calculate_bmi()
    db.session.add(record)
    db.session.commit()
    flash('Health record added!', 'success')
    return redirect(url_for('health.view'))


@health_bp.route('/plan/<int:id>/complete', methods=['POST'])
def complete_plan_item(id):
    item = HealthPlanItem.query.get_or_404(id)
    item.completed = not item.completed
    db.session.commit()
    return redirect(url_for('health.view'))


@health_bp.route('/plan/<int:id>/delete', methods=['POST'])
def delete_plan_item(id):
    item = HealthPlanItem.query.get_or_404(id)
    db.session.delete(item)
    db.session.commit()
    flash('Plan item deleted.', 'info')
    return redirect(url_for('health.view'))


@health_bp.route('/export')
def export_data():
    records = HealthRecord.query.order_by(HealthRecord.date.desc()).all()
    latest = HealthRecord.query.order_by(HealthRecord.date.desc()).first()

    content = "Health Data Export\n"
    content += "=" * 40 + "\n\n"
    if latest:
        content += f"Current Profile:\n"
        content += f"  Weight: {latest.weight} kg\n"
        content += f"  Height: {latest.height} cm\n"
        content += f"  BMI: {latest.bmi}\n"
        content += f"  Body Fat: {latest.body_fat}%\n\n"

    content += "Records:\n"
    content += "-" * 40 + "\n"
    for r in records:
        content += f"\nDate: {r.date}\n"
        content += f"  Weight: {r.weight} kg\n"
        content += f"  Height: {r.height} cm\n"
        content += f"  BMI: {r.bmi}\n"
        content += f"  Body Fat: {r.body_fat}%\n"

    return Response(
        content,
        mimetype='text/plain',
        headers={'Content-Disposition': 'attachment; filename=health_data.txt'}
    )
